pages/auth/login/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect } from "react";
+import {
+  AuthContainer,
+  LogoHomeLink,
+  Form,
+  AuthSwitch,
+  SubmitButton,
+} from "../AuthStyles";
+import Logo from "@/public/images/logo/logo.svg";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import InputItem from "@/components/ui/InputItem";
+import SocialLogin from "../components/SocialLogin";
+import PasswordInput from "../components/PasswordInput";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { LoginFormValues } from "@/types/authTypes";
+import { requestLogin } from "@/api/authApi";
+import Image from "next/image";
+
+const LoginPage: React.FC = () => {
+  // 답안 해설은 회원가입 페이지 컴포넌트를 참고해 주세요!
+
+  const router = useRouter();
+
+  // 만약 로그인 컴포넌트 마운트 시 localStorage에 accessToken가 존재한다면 이미 로그인 상태라는 의미이므로 홈페이지로 리라우팅
+  useEffect(() => {
+    const accessToken = localStorage.getItem("accessToken");
+    if (accessToken) {
+      router.push("/");
+    }
+  }, [router]);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    trigger,
+    formState: { errors, isValid },
+  } = useForm<LoginFormValues>({ mode: "onChange" });
+
+  const onSubmit: SubmitHandler<LoginFormValues> = async (data) => {
```

Comment:
P2:
data 자체로 사용하지 않으니 아래처럼 분해해서 쓰시는게 더 좋을 것 같습니다.

```suggestion
  const onSubmit: SubmitHandler<LoginFormValues> = async ({ email, password }) => {
```